### PR TITLE
store: convert request error to warning when cop request to tidb

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

If there is a TiDB panic in the cluster, when query cluster table like `cluster_slow_query`, `cluster_processlist` will got error like `TiKV server timeout`. the usability is not good, convert the error to waning is better, and keep this behavior consistant with other cluster tables, such as `cluster_log`, `cluster_config`...

Before this PR:

```sql
mysql>select count(*) from `CLUSTER_SLOW_QUERY`;
(9002, u'TiKV server timeout')
```

This PR

```sql
mysql>select count(*) from `CLUSTER_SLOW_QUERY`;
+----------+
| count(*) |
+----------+
| 96432    |
+----------+
1 row in set
Time: 40.771s
mysql>show warnings;
+---------+------+-------------------------------------------------+
| Level   | Code | Message                                         |
+---------+------+-------------------------------------------------+
| Warning | 9002 | TiDB server timeout, address is 127.0.0.1:10081 |
+---------+------+-------------------------------------------------+
```

### What is changed and how it works?

Convert error to warning when the cop request is send to `TiDB`.

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

```sql
-- start a cluster with 2 tidb-server
-- kill one of the tidb-server
select count(*) from `CLUSTER_SLOW_QUERY`
```

### CODE REVIEW

File: store/tikv/coprocessor.go [hunk 1]
```diff
@@ -1003,6 +1011,35 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *RPCCon
...
+		Warnings: []*tipb.Error{
+			{
+				Code: int32(errCode),
+				Msg:  errMsg,
+			},
+		},
+	}
+	data, err := proto.Marshal(&selResp)
+	if err != nil {
+		return nil
```
Comment (tiancaiamao) [inline store/tikv/coprocessor.go]: return errors.Trace(err)

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -1003,6 +1011,35 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *RPCCon
...
+	}
+	resp := &copResponse{
+		pbResp: &coprocessor.Response{
+			Data: data,
+		},
+		detail: &execdetails.ExecDetails{},
+	}
+	worker.sendToRespCh(resp, ch, true)
```
Comment (tiancaiamao) [inline store/tikv/coprocessor.go]: Is there any change that the goroutine block on here forever?

File: store/tikv/coprocessor.go [hunk 3]
```diff
@@ -1003,6 +1011,35 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *RPCCon
 	return nil, nil
 }
 
+func (worker *copIteratorWorker) handleTiDBSendReqErr(err error, task *copTask, ch chan<- *copResponse) error {
+	errCode := 0
+	errMsg := err.Error()
+	if terror.ErrorEqual(err, ErrTiKVServerTimeout) {
```
Comment (tiancaiamao) [inline store/tikv/coprocessor.go]: Could the error be something else other than `ErrTiKVServerTimeout`?
errCode = 0 seems not a good default, how about using unknown error code?

File: store/tikv/coprocessor.go (hunk 1 above)
Comment (crazycs520) [reply store/tikv/coprocessor.go]: great, done.

File: store/tikv/coprocessor.go (hunk 3 above)
Comment (crazycs520) [reply store/tikv/coprocessor.go]: great, done.

File: store/tikv/coprocessor.go (hunk 2 above)
Comment (crazycs520) [reply store/tikv/coprocessor.go]: I think there is no related change that may block the channel here.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**